activejob/test/cases/railtie_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "helper"
+require "active_support/testing/isolation"
+
+class RailtieTest < ActiveSupport::TestCase
+  include ActiveSupport::Testing::Isolation
+
+  setup do
+    require "rails"
+
+    @rails_logger = Logger.new(nil)
+
+    @app ||= Object.const_set(
+      "AJRailtieTestApp#{SecureRandom.alphanumeric(6)}",
+      Class.new(::Rails::Application)
+    ) do |app|
+      config.eager_load = false
+      config.logger = @rails_logger
+    end
+  end
+
+  test "active_job.logger initializer does not overwrite the supplied logger" do
+    custom_logger = Logger.new(nil)
+
+    @app.config.before_initialize do |app|
+      ActiveSupport.on_load(:active_job) do
+        self.logger = custom_logger
+      end
+    end
+
+    require "active_job/railtie"
+    @app.initialize!
+
+    assert_equal true, ActiveJob::Base.logger.object_id == custom_logger.object_id
```

Comment:
We can use `assert_same` here.

Reply:
[Use assert_same instead of assert_equal](https://github.com/rails/rails/pull/50164/commits/6da5a924053ea79947990b4daf9fe26e26a42153)

Reply:
For small PRs, you can just force push and keep commits to 1. So committers won't ask you to squash everything, when merging.

Reply:
```ruby
assert_same custom_logger, ActiveJob::Base.logger
```
no need to test for `object_id` be the same (`assert_same` does this internally)